.circleci/config.yml

```diff
@@ -0,0 +1,73 @@
+version: 2
+jobs:
+  lint:
+    docker:
+      - image: hadolint/hadolint:v1.17.2-debian
+    steps:
+      - checkout
+      - run:
+          name: Lint dockerfile
+          command: ./scripts/lint.sh
+  build:
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: |
+            docker build -t zenika/terraform-azure-cli:latest-debian -f debian.Dockerfile .
+            docker build -t zenika/terraform-azure-cli:latest-alpine -f alpine.Dockerfile .
+  build-tag:
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: |
+            docker build -t zenika/terraform-azure-cli:$CIRCLE_TAG-debian -f debian.Dockerfile .
+            docker build -t zenika/terraform-azure-cli:$CIRCLE_TAG-alpine -f alpine.Dockerfile .
+  publish-latest:
+    environment:
+      IMAGE_NAME: building-on-ci
+    context: docker-hub
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    steps:
+      - setup_remote_docker
+      - run:
+          name: Publish Docker Image to Docker Hub
+          command: |
+            echo "$DOCKER_HUB_PASSWORD" | docker login -u "$DOCKER_HUB_USER" --password-stdin
+            docker push zenika/terraform-azure-cli:latest-alpine
+            docker push zenika/terraform-azure-cli:latest-debian
+            docker push zenika/terraform-azure-cli:$CIRCLE_TAG-alpine
+            docker push zenika/terraform-azure-cli:$CIRCLE_TAG-debian
+workflows:
+  version: 2
+  test-and-build:
+    jobs:
+      - lint
+      - build:
+          requires:
+            - lint
+      - build-tag:
+          requires:
+            - lint
+          filters:
+            tags:
+              only: /v?[0-9]+\.[0-9]+/
+            branches:
+              ignore: /.*/
+      - publish:
+          requires:
+            - build
+            - build-tag
+          filters:
+            tags:
+              only: /v?[0-9]+\.[0-9]+/
```

Comment:
I think we can get rid of the v in the tag, to folow the current project tag convention 😉